.editorconfig

```diff
@@ -1,6 +1,371 @@
+#   generate conditional expression over return,
+#   suggest inferred anonymous types,
+#   suggest inferred tuple names,
+#   suggest 'is null' checks over '== null',
+#   don't generate 'this.' and 'Me.' for events,
+#   warn when not using 'this.' and 'Me.' for fields,
+#   warn when not using 'this.' and 'Me.' for methods,
+#   warn when not using 'this.' and 'Me.' for properties,
+#   suggest readonly fields, and
+#   generate accessibility modifiers for non interface members
+###############################################################################
+[*.cs]
+dotnet_style_coalesce_expression = true:suggestion
+dotnet_style_collection_initializer = true:suggestion
+dotnet_style_explicit_tuple_names = true:suggestion
+dotnet_style_null_propagation = true:suggestion
+dotnet_style_object_initializer = true:suggestion
+
+dotnet_style_parentheses_in_arithmetic_binary_operators = always_for_clarity:silent
+dotnet_style_parentheses_in_other_binary_operators = always_for_clarity:silent
+dotnet_style_parentheses_in_other_operators = never_if_unnecessary:silent
+dotnet_style_parentheses_in_relational_binary_operators = always_for_clarity:silent
+
+dotnet_style_predefined_type_for_locals_parameters_members = true:silent
+dotnet_style_predefined_type_for_member_access = true:silent
+
+dotnet_style_prefer_auto_properties = true:silent
+dotnet_style_prefer_compound_assignment = true:suggestion
+dotnet_style_prefer_conditional_expression_over_assignment = true:silent
+dotnet_style_prefer_conditional_expression_over_return = true:silent
+dotnet_style_prefer_inferred_anonymous_type_member_names = true:suggestion
+dotnet_style_prefer_inferred_tuple_names = true:suggestion
+dotnet_style_prefer_is_null_check_over_reference_equality_method = true:suggestion
+
+dotnet_style_qualification_for_event = false:silent
+dotnet_style_qualification_for_field = false:silent
+dotnet_style_qualification_for_method = false:silent
+dotnet_style_qualification_for_property = false:silent
+
+dotnet_style_readonly_field = true:suggestion
+dotnet_style_require_accessibility_modifiers = for_non_interface_members:silent
+
+###############################################################################
+# Set dotnet style options to:
+#   suggest removing all unused parameters
+###############################################################################
+[*.cs]
+dotnet_code_quality_unused_parameters = all:suggestion
+
+###############################################################################
+# Set csharp indent options to:
+#   indent block contents,
+#   not indent braces,
+#   indent case contents,
+#   not indent case contents when block,
+#   indent labels one less than the current, and
+#   indent switch labels
+###############################################################################
+[*.cs]
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents = true
+csharp_indent_case_contents_when_block = false
```

Comment:
This is the only new option that differs from the VS default. The VS default is `true` for compatibility reasons. This option is used in combination with `csharp_indent_case_contents` and you have effectively the following combinations:

(VS Default) `csharp_indent_case_contents=true`, `csharp_indent_case_contents_when_block=true`:
```cs
switch (value)
{
    case 1:
        break;

    case 2:
        {
            break;
        }
}
```

`csharp_indent_case_contents=false`, `csharp_indent_case_contents_when_block=true`:
```cs
switch (value)
{
    case 1:
    break;

    case 2:
        {
            break;
        }
}
```

`csharp_indent_case_contents=false`, `csharp_indent_case_contents_when_block=false`:
```cs
switch (value)
{
    case 1:
    break;

    case 2:
    {
        break;
    }
}
```

(The one I chose) `csharp_indent_case_contents=true`, `csharp_indent_case_contents_when_block=false`:
```cs
switch (value)
{
    case 1:
        break;

    case 2:
    {
        break;
    }
}
```

Reply:
I personally don't like the corefx style, but since you've written most of the code here now you can pick whatever you want.

Reply:
Which one of the four is your preference? I don't feel particularly strongly as long as the editorconfig ensures it is automatically normalized.

Reply:
The last one seems ok. I don't know why it's not the default. The VS Default looks ugly.

Reply:
The last one is the one I already choose for the repo (and was the one I thought you were indicating you didn't like) 😄

The first is the VS default for compatibility reasons at this point. The option allowing the last setup was only introduced a couple versions ago (I think VS2015 or 2017) and hasn't been around for very long at this point.